java/component/keyple-plugin/remote-se/nativese/src/main/java/org/eclipse/keyple/plugin/remotese/nativese/impl/TransmitSetExecutor.java

```diff
@@ -0,0 +1,107 @@
+
+/**
+ * Execute locally a TRANSMIT KeypleMessageDto
+ */
+class TransmitSetExecutor implements Executor {
+
+    private static final Logger logger = LoggerFactory.getLogger(TransmitSetExecutor.class);
+
+    private final ProxyReader reader;
+
+    /**
+     * Builds a TRANSMIT KeypleMessageDto executor
+     */
+    TransmitSetExecutor(ProxyReader reader) {
+        this.reader = reader;
+    }
+
+    @Override
+    public KeypleMessageDto execute(KeypleMessageDto keypleMessageDto) {
+        List<SeResponse> seResponses = null;
+        MultiSeRequestProcessing multiSeRequestProcessing;
+        ChannelControl channelControl;
+        KeypleMessageDto response = null;
+
+        // parse body
+        JsonObject bodyJsonO =
+                KeypleJsonParser.getParser().fromJson(keypleMessageDto.getBody(), JsonObject.class);
+
+        // extract info
+        multiSeRequestProcessing = MultiSeRequestProcessing
+                .valueOf(bodyJsonO.get("multiSeRequestProcessing").getAsString());
+
+        channelControl = ChannelControl.valueOf(bodyJsonO.get("channelControl").getAsString());
+
+        List<SeRequest> seRequests =
+                KeypleJsonParser.getParser().fromJson(bodyJsonO.get("seRequests").getAsString(),
+                        new TypeToken<ArrayList<SeRequest>>() {}.getType());
+
+
+        // prepare transmitSet on nativeReader
+        logger.trace("Execute locally seRequests : {} with params {} {}", seRequests,
+                channelControl, multiSeRequestProcessing);
+
+
+        try {
+
+            // execute transmitSet
+            seResponses =
+                    reader.transmitSeRequests(seRequests, multiSeRequestProcessing, channelControl);
+
+            // prepare response body
+            String body = KeypleJsonParser.getParser().toJson(seResponses,
+                    new TypeToken<ArrayList<SeResponse>>() {}.getType());
+
+            // build response Dto
+            response = new KeypleMessageDto(keypleMessageDto).setBody(body);;
+
+        } catch (KeypleReaderIOException e) {
+
+            JsonObject body = new JsonObject();
+            if (e.getSeResponses() != null) {
+                body.add("seResponses", KeypleJsonParser.getParser().toJsonTree(e.getSeResponses(),
+                        new TypeToken<List<SeResponse>>() {}.getType()));
+            }
+            if (e.getSeResponse() != null) {
+                body.add("seResponse", KeypleJsonParser.getParser().toJsonTree(e.getSeResponse(),
+                        SeResponse.class));
+            }
+
+            // if an exception occurs, send it into a keypleDto to the Master
+            response = new KeypleMessageDto(keypleMessageDto)
+                    .setAction(KeypleMessageDto.Action.ERROR.name())//
+                    .setErrorCode(KeypleMessageDto.ErrorCode.KeypleReaderIOException.getCode())//
+                    .setErrorMessage(e.getMessage())//
+                    .setBody(body.toString());
```

Comment:
Même remarque

Reply:
KeypleReaderIoException contains SeResponses, then we must send them to the server.